.github/maven.yml

```diff
@@ -0,0 +1,58 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: GitHub CI
+
+on: workflow_call
+  inputs:
+    maven_args:
+      description: The arguments to pass to Maven when building the code
+      required: false
+      default: --batch-mode --errors --show-version verify
+      type: string
+
+jobs:
+  build:
+
+    strategy:
+      matrix:
+        os: [ ubuntu-latest,windows-latest, macOS-latest ]
+        java: [ 8, 11, 16, 17-ea ]
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up cache for ~./m2/repository
```

Comment:
setup-java now have a cache (https://github.com/actions/setup-java#caching-maven-dependencies) maybe we can this step now.